### PR TITLE
Make {NotificationHistory,StateHistory,History*}#Id UUID -> SHA1

### PR DESCRIPTION
**Pull request by Al2Klimov**
## Blocked by

* Icinga/icinga2#9029

### CODE REVIEW

**Comment by Al2Klimov** [comment]
```
*************************** 14. row ***************************
                        id: 0x767E1B079E329D3D9CBB84E986B53F2E1EE0F43A
            environment_id: 0xDA39A3EE5E6B4B0D3255BFEF95601890AFD80709
               endpoint_id: NULL
               object_type: host
                   host_id: 0x5FFD3470E558666B25CCD505C329FAA0F1932596
                service_id: NULL
   notification_history_id: 0x767E1B079E329D3D9CBB84E986B53F2E1EE0F43A
          state_history_id: NULL
       downtime_history_id: NULL
        comment_history_id: NULL
       flapping_history_id: NULL
acknowledgement_history_id: NULL
                event_type: notification
                event_time: 1634290165111
*************************** 15. row ***************************
                        id: 0x78AFE8D478F7F2CE6C030B7940B07BA68A0E102E
            environment_id: 0xDA39A3EE5E6B4B0D3255BFEF95601890AFD80709
               endpoint_id: NULL
               object_type: service
                   host_id: 0x5FFD3470E558666B25CCD505C329FAA0F1932596
                service_id: 0x1BD19F3BB92A95F39A3D05B7E5C682500250591B
   notification_history_id: NULL
          state_history_id: 0x78AFE8D478F7F2CE6C030B7940B07BA68A0E102E
       downtime_history_id: NULL
        comment_history_id: NULL
       flapping_history_id: NULL
acknowledgement_history_id: NULL
                event_type: state_change
                event_time: 1634290138562
```

```
*************************** 3. row ***************************
                 id: 0xCF38DC68CE576C29DD9642EE923EA722D934F696
     environment_id: 0xDA39A3EE5E6B4B0D3255BFEF95601890AFD80709
        endpoint_id: NULL
        object_type: host
            host_id: 0x5FFD3470E558666B25CCD505C329FAA0F1932596
         service_id: NULL
    notification_id: 0xC9B5E210E1891F369677D711A6BF84A9C268591E
               type: problem
          send_time: 1634288049928
              state: 1
previous_hard_state: 99
             author:
               text: /sbin/ping6 -n -c 5 127.0.0.1
CRITICAL - Could not interpret output from ping command
     users_notified: 1
```

```
*************************** 1. row ***************************
                     id: 0xA573669A53E2145D2AA523D507C90F84317D3114
         environment_id: 0xDA39A3EE5E6B4B0D3255BFEF95601890AFD80709
notification_history_id: 0x18880C842A30AAB26CC5186FCD3C5322B2F0FEE4
                user_id: 0x25810609DC85D7733543B8EC885580C111B45EA3
```

File: schema/mysql/schema.sql [hunk 1]
```diff
@@ -916,7 +916,7 @@ CREATE TABLE zone (
 ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin ROW_FORMAT=DYNAMIC;
 
 CREATE TABLE notification_history (
-  id binary(16) NOT NULL COMMENT 'UUID',
+  id binary(20) NOT NULL COMMENT 'sha1(environment.name + notification.name + send_time)',
```
Comment (julianbrost) [inline schema/mysql/schema.sql:919]: Misses notification type which is now also included in the hash.

**Review by julianbrost — CHANGES_REQUESTED** [review]
Just one minor issue in a schema comment, looks fine apart from that, so please set the Redis schema version to 2 for this set of PRs.

File: schema/mysql/upgrades/1.0.0-rc2.sql [hunk 2]
```diff
@@ -173,10 +174,17 @@ ALTER TABLE service_state
     MODIFY long_output longtext DEFAULT NULL,
     MODIFY performance_data longtext DEFAULT NULL;
 
+ALTER TABLE notification_history
+    MODIFY id binary(20) NOT NULL COMMENT 'sha1(environment.name + notification.name + send_time)';
```
Comment (julianbrost) [inline schema/mysql/upgrades/1.0.0-rc2.sql:178]: The change adding type is missing here.

**Review by julianbrost — CHANGES_REQUESTED** [review]

File: schema/mysql/upgrades/1.0.0-rc2.sql (hunk 2 above)
Comment (Al2Klimov) [reply schema/mysql/upgrades/1.0.0-rc2.sql:178]: Don’t say this file still exists?

Comment (julianbrost) [reply schema/mysql/upgrades/1.0.0-rc2.sql:178]: At the moment, it's still there, and as long as it is, I'd still keep it up-to-date.

Comment (Al2Klimov) [reply schema/mysql/upgrades/1.0.0-rc2.sql:178]: ... unless it's gone in the **next** moment.

Comment (julianbrost) [reply schema/mysql/upgrades/1.0.0-rc2.sql:178]: Come on, the change is trivial. At the moment, the PR still updates the upgrade file, but in a way that doesn't match the schema change. Do you mind if I just push a commit adding " + type" here too?

**Review by julianbrost — APPROVED** [review]

**Pull request merged**